cors/docs/cloudshell-tutorial.md

```diff
@@ -0,0 +1,75 @@
+# CORS Security
+
+---
+This sample lets you create an API that uses the cross-origin resource sharing (CORS) mechanism to allow requests from external webpages and applications
+
+Let's get started!
+
+---
+
+## Setup instructions
+
+1. Navigate to the 'cors' directory in the Cloud Shell.
+
+```sh
+cd cors
+```
+
+2. Ensure you have an active GCP account selected in the Cloud shell
+
+```sh
+gcloud auth login
+```
+
+3. Edit the `env.sh` and configure the ENV vars. Click <walkthrough-editor-open-file filePath="cors/env.sh">here</walkthrough-editor-open-file> to open the file in the editor
+
+* `PROJECT` the project where your Apigee organization is located
+* `APIGEE_HOST` the externally reachable hostname of the Apigee environment group that contains APIGEE_ENV
+* `APIGEE_ENV` the Apigee environment where the demo resources should be created
+
+Now source the `env.sh` file
+
+```bash
+source ./env.sh
+```
+
+## Deploy Apigee components
+
+Next, let's deploy some CORS protected Apigee sample proxy
+
+```bash
+./deploy-cors.sh
+```
+
+---
+
+## Test CORS Proxy
+
+Now that our API proxy is deployed, let's enable the debugger so we can see requests as they come through:
+1. Navigate to the [Apigee homepage](https://apigee.google.com). Go to Develop > API Proxies and click into the sample-cors proxy
+2. Navigate to the debug tab
+3. Choose to start a debug session for the currently deployed proxy version
+
+With the proxy debugger still running, call the sample CORS proxy from a cross-origin client:
+1. Navigate to an online HTTP request service such as [test-cors.org](https://test-cors.org). The requests need to be sent from an online service and should not be sent from your local machine.
+2. From [test-cors.org](https://test-cors.org) find the Remote URL field and enter in your proxy's URL, https://\[APIGEE_HOST\]/v1/samples/cors. Leave all other fields at their default values
+3. Click the Send Request button and view the response. You should see 200 response, meaning our CORS policy is working as expected and our test was a success!
+4. Navigate to the Apigee debugger. You should see a 200 response there as well!
```

Comment:
Maybe it's worth showing an example of a blocked request as well.  What do you think?

Reply:
I thought about this, and talked about it with some others as well. The decision we came to was to keep the accelerator as short and sweet as possible. Perhaps I could add a "Optional: Trigger CORS Error" section at the end?

Reply:
If you're going to add a failed request, I think it should come before the successful one.  I think having both a negative and a positive result helps to demonstrate that the policy is really doing what it's supposed to.